src/systems/si-hep/include/units/isq/si/hep/mass.h

```diff
@@ -44,7 +44,8 @@ namespace units::isq::si::hep {
 
 struct eV_per_c2 :
     named_scaled_unit<eV_per_c2, basic_symbol_text{"eV/c²", "eV/c^2"},
-                      as_magnitude<ratio(17'826'619'216'279, 1'000'000'000'000, -35)>(), kilogram> {};
+                      as_magnitude<ratio(17'826'619'216'279, 1'000'000'000'000)>() * pow<-35>(as_magnitude<10>()),
```

Comment:
It seems that `pow<-35>(as_magnitude<10>())` is a common pattern. Can we simplify this or provide some helper (i.e. `as_magnitude<10, -35>`)?

Reply:
Also, there could be an `operator*(ratio, magnitude)` (and the inverse) that makes the magnitude from the other ratio as well. With that, we could have something like:

```cpp
ratio(17'826'619'216'279, 1'000'000'000'000) * as_magnitude<10, -35>()
```

Reply:
Right :-(

Reply:
`as_magnitude<I, J>` feels ambiguous to my eye.  I might be tempted to interpret it as $I/J$ rather than $I^J$.

WDYT about `mag_power<I, J>`, where `I` is anything that can convert to `ratio`, and will usually be an integer?

Reply:
Yes, `I` should be `convertible_to<ratio>.

Reply:
@chiphogg, what do you think about such an interface? I think it would simplify a lot of unit definitions which is the end-user interface thus should be as simple as possible. If you do not like it, we can merge the PR as it is now and possibly refactor the code later. Although, now is probably easier to find all the places that need such refactoring.

Reply:
I love the interface, I'm just not sure about the name.  At callsites, `as_magnitude<I, J>` looks too much like a ratio, $I/J$.  That's the paradigm that will be most familiar to people when they see `<int, int=1>` interfaces, thanks to experience with the ubiquitous `std::ratio`.

One alternative I like is `mag_power<ratio, ratio>`.  I like it even better if we rename the somewhat unwieldy `as_magnitude` to `mag`, as it now is in Aurora's library.  Then we have:

- `mag<ratio>`
- `mag_power<ratio, ratio>`

That said, I'm open to other suggestions, and even to `as_magnitude<ratio, ratio>` if you think my concerns are unfounded.  We really do need a simpler interface for the callsites!

Reply:
Do you prefer doing the refactoring in this or a new PR? In the latter case, I can merge this one ASAP.

Reply:
I'll do the `mag_power` in this one, since it'll be easy to find all the places, but doing `mag` here would explode the size of this PR.  I think that'll be cleaner in its own PR.